File: 2-year/OOP-CSharp/src/Lab5/Presentation/Lab5.Presentation.Console/Scenarios/PutMoney/PutMoneyScenario.cs
using Lab5.Application.Contracts.Users;
using Lab5.Application.Contracts.Users.OperationsResults;
using Spectre.Console;

namespace Lab5.Presentation.Console.Scenarios.PutMoney;

public class PutMoneyScenario : IScenario
{
    private readonly IUserService _userService;

    public PutMoneyScenario(IUserService userService)
    {
        _userService = userService;
    }

    public string Name => "Put money";

    public void Run()
    {
        long accountId = AnsiConsole.Ask<long>("Enter your account id: ");
        decimal amount = AnsiConsole.Ask<decimal>("Enter money amount: ");

        PutMoneyResult result = _userService.PutMoney(accountId, amount);

        string message = result switch
        {
            PutMoneyResult.Success => "Success!",
            PutMoneyResult.AccountNotFound => "Account not found",
            PutMoneyResult.NotLoginYet => "Please login to continue",
            PutMoneyResult.NegativeAmountOfMoney => "Amount of money cannot be negative",
            _ => throw new ArgumentOutOfRangeException(nameof(result)),
        };

        AnsiConsole.WriteLine(message);
        AnsiConsole.Ask<string>("Ok");
    }
}

File: 2-year/Java/Lab1/src/main/java/com/Starygin/Presentation/Scenarios/PutMoneyScenario.java
package com.Starygin.Presentation.Scenarios;

import com.Starygin.Application.Contracts.UserService;
import com.Starygin.Presentation.Scenario;

import java.io.IOException;
import java.util.Scanner;

public class PutMoneyScenario implements Scenario {
    private final UserService userService;

    public PutMoneyScenario(UserService userService) {
        this.userService = userService;
    }

    @Override
    public String getName() {
        return "Put Money";
    }

    @Override
    public void run() {
        System.out.println("Enter account id: ");
        Scanner scanner = new Scanner(System.in);

        try {
            long accountId = scanner.nextLong();
            System.out.println("Enter money amount: ");
            double amount = scanner.nextDouble();

            userService.putMoney(accountId, amount);

            System.out.println("Successfully put money");

        } catch (IOException e) {
            System.out.println(e.getMessage());
        }
    }
}
